Add record&recordControler&PR Test

Kwak-min opened the pull request:
Add record
Add recordControler
PR Test

Kwak-min committed:
back-end

+ record
+ recordController

saickersj123 commented:
- Record가 정확히 어떤 기록을 의미하는지 알아볼 수 있도록 네이밍을 명시적으로 수정할 것
- 키워드 체크 방식 개선 필요, 각 스테이지 마다 키워드를 DB 스키마를 적절히 설계하여 저장하고 스테이지 마다 키워드를 불러와 비교 및 평가하는 식으로 개선

saickersj123 reviewed (changes requested):
- Record가 정확히 어떤 기록을 의미하는지 알아볼 수 있도록 네이밍을 명시적으로 수정 필요
- 키워드 체크 방식 개선 필요 -> 각 스테이지 마다 키워드를 DB 스키마를 적절히 설계하여 저장하고 스테이지 마다 키워드를 불러와 비교 및 평가.
- package-lock.json 변경 사항 확인해서 어떤 부분 변경인지 해당 댓글에 답글로 게시

Kwak-min commented:
package-lock.json 파일 경우 최신버전 사용을 위한 업데이트이며,
Record는 Schema이며 클리어 관련 정보 저장을 위한 저장 Schema입니다.
Recontroller의 경우 record를 사용하여 정보 조회 및 기록 추가 기록 삭제 등의 Controller입니다.

saickersj123 commented:
Packge lock의 어떤 패키지가 업데이트 된건지를 물어본것이고
코드 가독성, 확장성을 위해서 record가 아닌 play record 라던지 한눈에 알아볼 수 있게 수정하라는 말입니다
키워드 평가 방식 또한 하드코딩된 방식이 아닌 모듈성과 원자성을 띄도록 개선해서 추가 작업바랍니다

saickersj123 commented:
요청된 변경사항이 반영되면 해당 PR을 종료하겠습니다

Kwak-min committed:
Back-End

return package-lock.json
change name record-> GameRecord
change name recordController -> GameRecordController

Kwak-min commented:
키워드 평가 방식 또한 하드코딩된 방식이 아닌 모듈성과 원자성을 띄도록 개선해서 추가 작업
-> DB에서 가져오는 방식이 아닌 AI 판단에 키워드를 설정하게 하라는 말씀이신가요?

saickersj123 commented:
우선은 스테이지에 대한 모델 스키마 구성하고 키워드를 평가하는 로직에 맞게 응답마다 키워드 평가를 할 때 활용할 키워드들을 저장하고 평가시 불러오도록,키워드 평가에 대한 상세한 로직 설계도 필요, 응답 마다 평가 할 경우, 언급한대로 AI로 자동화하는게 효과적일 수 있음, AI가 아니여도 충분히 구현은 가능해보임

saickersj123 commented:
우선은 스테이지에 대한 모델 스키마 구성하고 키워드를 평가하는 로직에 맞게 응답마다 키워드 평가를 할 때 활용할 키워드들을 저장하고 평가시 불러오도록,키워드 평가에 대한 상세한 로직 설계도 필요, 응답 마다 평가 할 경우, 언급한대로 AI로 자동화하는게 효과적일 수 있음, AI가 아니여도 충분히 구현은 가능해보임

saickersj123 commented:
평가 기준 키워드는 스테이지를 등록할 때 함께 입력하도록해서 스테이지를 등록하는 모듈 또한 필요할 것, 스테이지 스키마에는 스테이지에 사용될 모델 정보 또한 포함되어야 함

saickersj123 commented:
해당 PR에는 일단 스테이지 스키마 만들고 DB에서 기준 키워드 불러오는 것까지만 구현하고 그 이후는 혼자하려 하지말고 팀원들과 함께 해결해보도록

saickersj123 commented:
package-lock은 github 웹에서 변경 사항 확인이 불가한데 해당 작업한 브랜치 또한 없어서 내용 확인이 아예 불가한 상황 변경 내용 게시 필요

Kwak-min commented:
package-lock 업데이트는 필요가 없어보여 undo했습니다.

Kwak-min committed:
Back-End

+ stage.ts
+ stageSchema

saickersj123 reviewed (approved):
스테이지 별 모델 데이터를 위한 스키마도 상세하게 설계 후, 구현하여 ref 또는 하위 종속하여 향후 업데이트 바라며, 어학 학습이라는 비즈니스 모델에 맞게 키워드 평가 뿐만 아니라 문법 검사 또한 추가 구현해야할 필요가 있어보입니다.
위에서 언급한 변경 사항은 다음 업데이트에 반영 바라며 해당 PR은 머지하고 종료하겠습니다.
